### PR TITLE
Old presence

### PR DESCRIPTION
**Pull request by rhyspatt**
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image

</details>

### CODE REVIEW

**Comment by EncryptedDev** [comment]
Hello @Sub2Rhys 👋

As I previously mentioned on the other PR:

> There are very few pages supported. If the service does not have any more pages, the service will not qualify for a persence at all

As there still aren't sufficient supported pages still, this pull request will be closed. 

Please **do not** reopen this pull request until a substantial number of pages has been added.

**Comment by Sub2Rhys** [comment]
Hey @EncryptedDev,

My presence is used to show what a user is looking at on the website, my website is for the game Fortnite which serves thousands of different cosmetics, all of which are on the website and have their own personal page. Although the base page for these items is the same, the data throughout each one changes providing the data for the presence to use.

[image](https://user-images.githubusercontent.com/59174048/171999177-39c0c9c8-c8d2-4c93-ace2-3fffb0c84316.png)
[image](https://user-images.githubusercontent.com/59174048/171999291-fe71ccd7-1836-4e71-a3d9-5e241eef7a58.png)

I also made my other pages follow a similar design to keep the code at a minimum, so each page contains elements with the 'itemImage', 'itemName' and 'itemDescription' ids.

With this, I have about 8 pages each containing hundreds of different items holding different pieces of data which can all be used in the presence itself.

[image](https://user-images.githubusercontent.com/59174048/171999508-375c8021-f24e-4d09-bf43-569ef1f9456e.png)
[image](https://user-images.githubusercontent.com/59174048/171999624-4994765e-6af2-4178-bb30-c46ef4cca5bf.png)
[image](https://user-images.githubusercontent.com/59174048/171999614-26d6c5ff-8a93-4a1b-8016-e46d3175fa96.png)

File: websites/R/RhysBOT/presence.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "Logo",
+			details: "Exploring",
+			startTimestamp: date,
+			buttons: [
+				{
+					label: "View Page",
+					url: document.location.href
```
Comment (ImRodry) [inline websites/R/RhysBOT/presence.ts:14]: you should add buttons when relevant and with appropriate labels, not on the initial declaration

File: websites/R/RhysBOT/presence.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+const presence = new Presence({
+		clientId: "658765364439810048"
+	}),
+	date = Date.now();
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "Logo",
```
Comment (ImRodry) [inline websites/R/RhysBOT/presence.ts:8]: keys should be camelCase

File: websites/R/RhysBOT/presence.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+			details: "Exploring",
```
Comment (ImRodry) [inline websites/R/RhysBOT/presence.ts:9]: this should be set in the code

File: websites/R/RhysBOT/presence.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+			startTimestamp: date,
+			buttons: [
+				{
+					label: "View Page",
+					url: document.location.href
+				}
+			]
+		},
+		page = document.location.pathname;
```
Comment (ImRodry) [inline websites/R/RhysBOT/presence.ts:18]: No reason to name it page, please change the variable to pathname elsewhere
```suggestion
		{ pathname } = document.location;
```

File: websites/R/RhysBOT/presence.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+			document.querySelector("#itemName").textContent
+		}`;
+
+		if (document.querySelector("#itemDescription")) {
+			presenceData.state = `${
+				document.querySelector("#itemDescription").textContent
+			}`;
+		}
+	} else if (page && !page.includes("_"))
+		presenceData.details = `Exploring the ${page} page`;
```
Comment (ImRodry) [inline websites/R/RhysBOT/presence.ts:34]: you should never put a pathname in the presenceData details like this. Please add checks for each individual page and customized messages

**Review by ImRodry — CHANGES_REQUESTED** [review]

**Comment by Sub2Rhys** [comment]
Hey @ImRodry, sorry about this but I'm having issues with the checks, it's picking up the other websites made by other people and causing the checks for my own website to fail, the only way to stop it from happening is to remove those websites from my own fork of the repository.

**Comment by ImRodry** [comment]
Please undo every miscellaneous change and rebase your branch properly

File: websites/F/Free Web Novel/tsconfig.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
...
-	"extends": "../../../tsconfig.json",
-	"compilerOptions": {
-	  "outDir": "./dist/"
-	}
+{
+	"extends": "../../../tsconfig.json",
+	"compilerOptions": {
+	  "outDir": "./dist/"
+	}
   }
```
Comment (ImRodry) [inline websites/F/Free Web Novel/tsconfig.json:6]: please remove unnecessary changes

File: websites/R/RhysBOT/presence.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
+const presence = new Presence({
+		clientId: "658765364439810048"
+	}),
+	date = Date.now();
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "logo",
+		details: "Exploring",
```
Comment (ImRodry) [inline websites/R/RhysBOT/presence.ts:9]: please only set presenceData details on specific pages or set them at the end of your code if nothing matches

File: websites/R/RhysBOT/presence.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+				document.querySelector("#itemDescription").textContent
+			}`;
+		}
+
+		presenceData.buttons[0].label = `View ${
+			document.querySelector("#itemName").textContent
+		}`;
+	}
+
+	if (presenceData.details) presence.setActivity(presenceData);
```
Comment (ImRodry) [inline websites/R/RhysBOT/presence.ts:38]: you always set presenceData.details so this will never be false

**Review by ImRodry — CHANGES_REQUESTED** [review]

File: websites/R/RhysBOT/dist/metadata.json [hunk 9]
```diff
@@ -0,0 +1,21 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "Rhys",
+		"id": "237674807561289731"
+	},
+	"service": "RhysBOT",
+	"description": {
+		"en": "An informative website for the world of fortnite."
```
[the remark this answers is not in the document]
Comment (Sub2Rhys) [reply websites/R/RhysBOT/dist/metadata.json:9]: image

I had it like that before...

File: websites/R/RhysBOT/presence.ts (hunk 7 above)
Comment (Sub2Rhys) [reply websites/R/RhysBOT/presence.ts:9]: okay

File: websites/F/Free Web Novel/tsconfig.json (hunk 6 above)
Comment (Sub2Rhys) [reply websites/F/Free Web Novel/tsconfig.json:6]: This is for 'Free Web Novel'?
